docker/geoserver/entrypoint.sh

```diff
@@ -3,82 +3,82 @@ set -e
 
 source /root/.bashrc
 
-# control the value of DOCKER_HOST_IP variable
-if [ -z ${DOCKER_HOST_IP} ]
-then
-
-    echo "DOCKER_HOST_IP is empty so I'll run the python utility \n"
-    echo export DOCKER_HOST_IP=`python3 /usr/local/tomcat/tmp/get_dockerhost_ip.py` >> /root/.override_env
-    echo "The calculated value is now DOCKER_HOST_IP='$DOCKER_HOST_IP' \n"
-
-else
 
-    echo "DOCKER_HOST_IP is filled so I'll leave the found value '$DOCKER_HOST_IP' \n"
-
-fi
+INVOKE_LOG_STDOUT=${INVOKE_LOG_STDOUT:-FALSE}
+invoke () {
+    if [ $INVOKE_LOG_STDOUT = 'true' ] || [ $INVOKE_LOG_STDOUT = 'True' ]
+    then
+        /usr/local/bin/invoke $@
+    else
+        /usr/local/bin/invoke $@ > /usr/src/geonode/invoke.log 2>&1
+    fi
+    echo "$@ tasks done"
+}
 
 # control the values of LB settings if present
-if [ ${GEONODE_LB_HOST_IP} ]
+if [ -n "$GEONODE_LB_HOST_IP" ];
 then
-
-    echo "GEONODE_LB_HOST_IP is filled so I replace the value of '$DOCKER_HOST_IP' with '$GEONODE_LB_HOST_IP' \n"
-    echo export DOCKER_HOST_IP=${GEONODE_LB_HOST_IP} >> /root/.override_env
-
+    echo "GEONODE_LB_HOST_IP is defined and not empty with the value '$GEONODE_LB_HOST_IP' \n"
+    echo export GEONODE_LB_HOST_IP=${GEONODE_LB_HOST_IP} >> /root/.override_env
+else
+    echo "GEONODE_LB_HOST_IP is either not defined or empty with the value 'django' \n"
+    echo export GEONODE_LB_HOST_IP=django >> /root/.override_env
 fi
 
-if [ ${GEONODE_LB_PORT} ]
+if [ -n "$GEONODE_LB_PORT" ];
 then
-
-    echo "GEONODE_LB_PORT is filled so I replace the value of '$PUBLIC_PORT' with '$GEONODE_LB_PORT' \n"
-    echo export PUBLIC_PORT=${GEONODE_LB_PORT} >> /root/.override_env
-
+    echo "GEONODE_LB_HOST_IP is defined and not empty with the value '$GEONODE_LB_PORT' \n"
+    echo export GEONODE_LB_PORT=${GEONODE_LB_PORT} >> /root/.override_env
+else
+    echo "GEONODE_LB_PORT is either not defined or empty with the value '8000' \n"
+    echo export GEONODE_LB_PORT=8000 >> /root/.override_env
 fi
 
 if [ ! -z "${GEOSERVER_JAVA_OPTS}" ]
 then
-
     echo "GEOSERVER_JAVA_OPTS is filled so I replace the value of '$JAVA_OPTS' with '$GEOSERVER_JAVA_OPTS' \n"
     JAVA_OPTS=${GEOSERVER_JAVA_OPTS}
-
 fi
 
 # control the value of NGINX_BASE_URL variable
 if [ -z `echo ${NGINX_BASE_URL} | sed 's/http:\/\/\([^:]*\).*/\1/'` ]
 then
     echo "NGINX_BASE_URL is empty so I'll use the static nginx hostname \n"
-    # echo export NGINX_BASE_URL=`python3 /usr/local/tomcat/tmp/get_nginxhost_ip.py` >> /root/.override_env
-    # TODO rework get_nginxhost_ip to get URL with static hostname from nginx service name
-    # + exposed port of that container i.e. http://geonode:80
-    echo export NGINX_BASE_URL=http://geonode:80 >> /root/.override_env
+    echo export NGINX_BASE_URL=http://django:8000 >> /root/.override_env
```

Comment:
You're right. Neither would work since this variable [is used inside Geoserver configuration](https://github.com/GeoNode/geonode-project/blob/align_with_geonode_master/docker/geoserver/set_geoserver_auth.sh#L62-L64) for the public URL of the LB (`Nginx`).

@afabiani We can revert this and sei it to "http://geonode:80" as it was before, although it's wrong.

Reply:
@giohappy yes, sure.

Reply:
I propose to set it to the echo GEOSERVER_LOCATION by default:

```bash
if [ -z `echo ${NGINX_BASE_URL} | sed 's/http:\/\/\([^:]*\).*/\1/'` ]
then
    echo "NGINX_BASE_URL is empty so I'll use the static nginx hostname \n"
    echo export NGINX_BASE_URL=http://${GEOSERVER_LOCATION} >> /root/.override_env
    echo "The calculated value is now NGINX_BASE_URL='${GEOSERVER_LOCATION}' \n"
else
    echo "NGINX_BASE_URL is filled so I'll leave the found value '$NGINX_BASE_URL' \n"
fi
```

Reply:
I see .. but what is the difference between `SITEURL` and `NGINX_BASE_URL`? Aren't both pointing to the same external URL? What infrastructure configuration do you have in mind so that the two variables would be different?

Reply:
@ridoo in environments like Azure the docker containers names are not resolved and therefore we need to provide here the ip address or some other internal reference.

Reply:
@afabiani I do not know Azure very much, so just fine with that. But which one is an internal config? As I understood @giohappy correctly, both point to the same value having just a different name in different containers.

Reply:
@ridoo genrally speaking this variable should **always** point to the `SITEURL` value, at least on a production like environment; if pointing to the containers (assuming that you started the NGINX one too) it must point to `geonode:80` otherwise it should point to whatever is currently proxying the `http(s)` requests.

Reply:
> I propose to set it to [GEOSERVER_LOCATION](https://github.com/GeoNode/geonode/blob/align_with_docker_master/.env.sample#L91) by default:
> 
> ```shell
> if [ -z `echo ${NGINX_BASE_URL} | sed 's/http:\/\/\([^:]*\).*/\1/'` ]
> then
>     echo "NGINX_BASE_URL is empty so I'll use the static nginx hostname \n"
>     echo export NGINX_BASE_URL=http://${GEOSERVER_LOCATION} >> /root/.override_env
>     echo "The calculated value is now NGINX_BASE_URL='${GEOSERVER_LOCATION}' \n"
> else
>     echo "NGINX_BASE_URL is filled so I'll leave the found value '$NGINX_BASE_URL' \n"
> fi
> ```
> 
> and we should probably do the reverse: set `GEOSERVER_LOCATION` to [$GEOSERVER_LOCATION](url) if `$NGINX_BASE_URL` is not set. It means: Geoserver location is the default one unless there's an LB in front of it.

@giohappy the `NGINX_BASE_URL` should be the `GEOSERVER_PUBLIC_LOCATION` not the `GEOSERVER_LOCATION`

Reply:
I've done this way @afabiani and @ridoo https://github.com/GeoNode/geonode-project/commit/3393da2c569353d62e5ffcc478853a46c5b3c297

Reply:
Sounds like a good idea. I scratched my head several times. But as you said: sometimes such things on their own slightly diverge from their original intent.